src/cast_core/src/CrudActor.cpp

```diff
@@ -1117,19 +1117,24 @@ struct CommitTransactionOperation : public BaseOperation {
 
 struct WithTransactionOperation : public BaseOperation {
     WithTransactionOperation(const Node& opNode,
-                             bool onSession,
+                             bool,
                              CollectionHandle collectionHandle,
                              metrics::Operation operation,
                              PhaseContext& context,
                              ActorId id)
         : BaseOperation(context, opNode),
-          _onSession{onSession},
           _collectionHandle{std::move(collectionHandle)},
           _operation{operation} {
+        auto& onSession = opNode["OnSession"];
+        if (!onSession || !onSession.to<bool>()){
+            BOOST_THROW_EXCEPTION(InvalidConfigurationException(
+                "'withTransaction' requires explicit 'OnSession: true'."));
```

Comment:
What is the reasoning behind this? If someone has explicitly set `OnSession=false`, then we should respect their choice and carry out the behavior exactly as requested. Why was this stricter route chosen instead of just setting it to true by default?

Does the YAML parsing system have a way to detect the difference between not being present and being false?

Reply:
(I'm guessing `!onSession` is undefined, the cast to bool is defined but false)

Reply:
> Does the YAML parsing system have a way to detect the difference between not being present and being false?

Yes. The `.maybe<T>()` gives you a `std::optional<T>` that can tell you if `{true, false, unspecified}`.

Reply:
- Indeed `!onSession` is the undefined check.
- [Transaction depends on session](https://www.mongodb.com/docs/manual/core/transactions-in-applications/#:~:text=When%20using%20the%20drivers%2C%20each%20operation%20in%20the%20transaction%20must%20be%20associated%20with%20the%20session%20(i.e.%20pass%20in%20the%20session%20to%20each%20operation).), so setting `OnSession=false` under `withTransaction` should be regarded as invalid configuration. @jimoleary suggested this and I agree with him.

Reply:
Okay, if it is going to generate a more difficult to diagnose error, then I agree, failing early here when we have context is the right choice. Happy to consider that resolved.

Where does this function get `onSession` from? Can that just become a `*bool` instead of going into manual mode here? I am uncomfortable with a function taking a parameter that it ignores and gets from somewhere else. This function needs to take all its YAML sourced values as args, or it needs to fetch them manually. I prefer the first if it can be done like that.

Reply:
`onSession` is passed in via the `OpCallback opCreator` variable [here](https://github.com/mongodb/genny/blob/3ca962077cb08b0f42b5e8eeba13616c5b66666a/src/cast_core/src/CrudActor.cpp#L1598-L1613). `opCreator` is the value in [this `opConstructors` unordered_map](https://github.com/mongodb/genny/blob/3ca962077cb08b0f42b5e8eeba13616c5b66666a/src/cast_core/src/CrudActor.cpp#L1221-L1245) that corresponds to the operation's name. `OpCallback` is declared [here](https://github.com/mongodb/genny/blob/3ca962077cb08b0f42b5e8eeba13616c5b66666a/src/cast_core/src/CrudActor.cpp#L139-L140).